src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -186,7 +186,7 @@ data::InstallationResult SotaUptaneClient::PackageInstallSetResult(const Uptane:
 void SotaUptaneClient::reportHwInfo() {
   Json::Value hw_info = Utils::getHardwareInfo();
   if (!hw_info.empty()) {
-    http->put(config.tls.server + "/core/system_info", hw_info);
+    http->put(config.tls.server + "/system_info", hw_info);
```

Comment:
@patrickvacek Perhaps it makes sense to introduce another configuration variable to set a device registry URL explicitly as in case of Director, ImageRepo and Treehub, config.tls.server is kind of opaque/vague - not clear to which of the backend services it points to.

Reply:
`config.tls.server` is the URL for the device gateway. We can rename it, but then we have to maintain backwards compatibility with the old name, which sounds tedious.

Reply:
I am not sure if backwards compatibility is required for real customers as this variable is basically automatically defined by aktualiz.
Also, there is inconsistency in the way how URLs are defined in the config, there are service specific URLs for all backend services except DeviceRegistry.

Reply:
> I am not sure if backwards compatibility is required for real customers as this variable is basically automatically defined by aktualiz.

We can't really guarantee that. If someone has overwritten `tls.server` in their config and then we change the name of the config variable in an update, their device will break and require manual intervention.

> there is inconsistency in the way how URLs are defined in the config, there are service specific URLs for all backend services except DeviceRegistry.

That's more or less abstracted from us. It depends on how the endpoints are defined in the device gateway, since that's the only service (other than treehub) that we actually talk to directly. We can't really do anything about that unless there is a specific path only used for forwarding to device registry, which I'm not sure is true. @simao or another backend dev may know better.

Reply:
> If someone has overwritten tls.server in their config and then we change the name of the config variable in an update, their device will break and require manual intervention.

I understand it and that's what I thought initially too, but on second thought I don't see an use case of `tls.server` server custom overriding by a customer that will work on practice unless they override it by the same value as it's set by default - DeviceGateway URL. Anyway, this is really not important at the moment.